src/univers/version_constraint.py

```diff
@@ -0,0 +1,221 @@
+
+        For example::
+        >>> assert VersionConstraint.split(">=2.3") == (">=", "2.3",)
+        >>> assert VersionConstraint.split("  <   =  2 . 3  ") == ("<=", "2.3",)
+        >>> assert VersionConstraint.split("2.3") == ("=", "2.3",)
+        >>> assert VersionConstraint.split("*2.3") == ("*", "",)
+        >>> assert VersionConstraint.split("*") == ("*", "",)
+        >>> assert VersionConstraint.split("<2.3") == ("<", "2.3",)
+        >>> assert VersionConstraint.split(">2.3") == (">", "2.3",)
+        >>> assert VersionConstraint.split("!=2.3") == ("!=", "2.3",)
+        """
+        constraint_string = remove_spaces(string)
+
+        # special case for star
+        if constraint_string.startswith("*"):
+            return "*", ""
+
+        for comparator in COMPARATORS:
+            if constraint_string.startswith(comparator):
+                # we do not report an error if this is not valid
+                version = constraint_string.lstrip("><=!")
+                if comparator == "*":
+                    version = ""
+                return comparator, version
+
+        # default to equality
+        return "=", constraint_string
+
+    # FIXME: this may be not enough to only handle "contains"?
+    def __contains__(self, version):
+        """
+        Return a True if the ``version`` Version is contained in this
+        VersionConstraint or "satisfies" this VersionConstraint.
+
+        For example::
+        >>> from univers.versions import PypiVersion
+        >>> v22 = PypiVersion("2.2")
+        >>> v23 = PypiVersion("2.3")
+        >>> v24 = PypiVersion("2.4")
+        >>> assert v23 in VersionConstraint(comparator="=", version=v23)
+        >>> assert v24 not in VersionConstraint(comparator="=", version=v23)
+        >>> assert None not in VersionConstraint(comparator="=", version=v23)
+
+        >>> assert None not in VersionConstraint(comparator="!=", version=v24)
+        >>> assert v22 in VersionConstraint(comparator="!=", version=v23)
+        >>> assert v23 in VersionConstraint(comparator="!=", version=v24)
+        >>> assert v24 not in VersionConstraint(comparator="!=", version=v24)
+
+        >>> assert v24 in VersionConstraint(comparator=">", version=v23)
+        >>> assert v23 not in VersionConstraint(comparator=">", version=v23)
+        >>> assert v24 in VersionConstraint(comparator=">=", version=v23)
+        >>> assert v23 in VersionConstraint(comparator=">=", version=v23)
+        >>> assert v22 not in VersionConstraint(comparator=">=", version=v23)
+
+        >>> assert v23 in VersionConstraint(comparator="<", version=v24)
+        >>> assert v23 in VersionConstraint(comparator="<=", version=v24)
+        >>> assert v24 in VersionConstraint(comparator="<=", version=v24)
+        >>> assert v24 not in VersionConstraint(comparator="<", version=v24)
+        """
+        if not version:
+            return False
+
+        # TODO: consider if we should accept any type and return False?
```

Comment:
This is debatable, and I have no clear answer.

Reply:
Raising a ValueError is best for now